agent/app/agent.go

```diff
@@ -108,6 +110,17 @@ func newAgent(
 		return nil, err
 	}
 
+	metadataManager := containermetadata.NewMetadataManager(dockerClient, cfg)
+	if !cfg.ContainerMetadataEnabled {
+		metadataManager = nil
+		log.Debugf("Container metadata feature disabled")
+	} else if metadataManager == nil {
+		// Terminate agent if container metadata is enabled but failed
+		log.Critical("Container metadata feature enabled but unable to start up. Terminating agent")
```

Comment:
If the feature is deliberately enabled, then it is likely the user will be running tasks that are dependent on the metadata reverse look-up. In the case, allowing a faulty agent to start-up will harm user productivity, whereas killing the agent will allow a restart and let a functioning Container Instance to run.